lang/java/reef-wake/wake/src/main/java/org/apache/reef/wake/time/runtime/RuntimeClock.java

```diff
@@ -97,80 +118,107 @@ public void scheduleAlarm(final int offset, final EventHandler<Alarm> handler) {
     }
   }
 
-  public void registerEventHandler(final Class<? extends Time> clazz, final EventHandler<Time> handler) {
-    this.handlers.subscribe(clazz, handler);
-  }
-
-  public void scheduleRuntimeAlarm(final int offset, final EventHandler<Alarm> handler) {
-    synchronized (this.schedule) {
-      this.schedule.add(new RuntimeAlarm(this.timer.getCurrent() + offset, handler));
-      this.schedule.notifyAll();
-    }
-  }
-
+  /**
+   * Stop the clock. Remove all other events from the schedule and fire StopTimer
+   * event immediately. It is recommended to use close() method for graceful shutdown
+   * instead of stop().
+   */
   @Override
   public void stop() {
     this.stop(null);
   }
 
+  /**
+   * Stop the clock on exception.
+   * Remove all other events from the schedule and fire StopTimer event immediately.
+   * @param exception Exception that is the cause for the stop. Can be null.
+   */
   @Override
-  public void stop(final Throwable stopOnException) {
-    LOG.entering(RuntimeClock.class.getCanonicalName(), "stop");
+  public void stop(final Throwable exception) {
+
+    LOG.entering(CLASS_NAME, "stop");
+
     synchronized (this.schedule) {
+
+      if (this.isClosed) {
+        LOG.log(Level.INFO, "Clock is already closed");
```

Comment:
This could become quite spammy on the logs. Maybe move to a more fine-grained log level?